### ISSUE TITLE
Unable to mount volumes to D: drive

### ISSUE THREAD

**Issue by VitorRigoni** [docker/for-win#1244]
Docker for windows doesn't seem to recognize drives other than C:/
I have a very small C:/ drive, so I set _Images and Volumes VHD Location_ to D:/ when I first installed and all was fine. 
Also, for any volumes I create I use my D drive.

After the fall creators update the path was reverted back to the default _C:\Users\Public\Documents\Hyper-V\Virtual Hard Disks\MobyLinuxVM.vhdx_ and when I create a volume in my D drive it creates it on the C drive (correct path and everything works fine, but wrong drive).

### Expected behavior
Volumes and MobyLinuxVM.vhdx are stored on D drive.

### Actual behavior
Volumes and MobyLinuxVM.vhdx are stored on C drive.

### Information
- Diagnostic ID: A460A434-C1CF-4047-A814-090860E952CC/2017-10-24_23-29-16
- Windows 10 Pro with Fall Creators Update

#### For Volumes
1. Run `docker run -v D:\\Docker:/usr/share/nginx/html -p 8080:80 -d nginx`
2. Add file `index.html` to `D:/Docker`
3. `curl http://localhost:8080` or `exec` into container and `cd /usr/share/nginx/html && ls -l`
4. Notice both curl fails and ls the html dir doesn't show the added index.html
5. Check volume folder was created on C drive.

#### For Virtual Hard Disk location
1. Open Settings -> Advanced
2. Change the _Images and Volumes VHD Location_ setting to D:/Docker/MobyLinuxVM.vhdx
3. Hit apply
4. Notice the input will revert back to the default value

**Comment by VitorRigoni** [comment]
Update:

For Virtual Disk Location I actually found this entry on the log:
```
[23:24:58.629][Linux          ][Warning] Migration failed with ClientSideFailedMessage: Migration operation on 'MobyLinuxVM' failed., ErrorSummaryDescription: Storage migration for virtual machine 'MobyLinuxVM'failed.

Destination VHD's GUID 'D:\Docker\MobyLinuxVM.vhdx' is different from the source VHD's GUID., ErrorDetailsDescription: Storage migration for virtual machine 'MobyLinuxVM' (4E04480A-13AE-473F-BE80-A4D5437C783A) failed with error 'Cannot create a file when that file already exists.' (0x800700B7).

Migration did not succeed. Destination VHD's GUID 'D:\Docker\MobyLinuxVM.vhdx' is different from the source VHD's GUID.
```

So deleting the existing file on D drive fixed the problem. The issue with the volumes, however, still persists.

**Comment by alexvy86** [comment]
@VitorRigoni's solution for the location of the `MobyLinuxVM.vhdx file` worked for me as well. Here we are trying to be nice to Docker, copying its file by hand before changing the path in Settings so it won't complain, and it "silently" complains and reverts to its default behavior :/.

**Comment by VitorRigoni** [comment]
Update:
I seem to have found the problem (and the solution).

For some reason, after the fall creator's update all shared drivers got messed up. I noticed that my C drive was being shared under the name "D".
Docker behavior was actually correct, while windows messed things up.

I removed all sharing settings from both drives via windows explorer as well as on docker settings and shared the D drive with docker once again. This fixed the issue.

Bottom line, looks like Windows messed up. Bad windows.
Closing the issue.
